valkeyrie.go

```diff
@@ -27,6 +29,9 @@ var (
 
 // NewStore creates an instance of store
 func NewStore(backend store.Backend, addrs []string, options *store.Config) (store.Store, error) {
+	initializersMutex.Lock()
+	defer initializersMutex.Unlock()
```

Comment:
Since we're using a `RWMutex` this could be a `RLock/RUnlock` here.

Or we could change to use a simple `Mutex` for `initializersMutex` instead since we're only using `Lock/Unlock`.